python/proton/_message.py

```diff
@@ -90,13 +90,18 @@ def _check(self, err):
 
     def _check_property_keys(self):
         for k in self.properties.keys():
-            if isinstance(k, unicode):
-                # py2 unicode, py3 str (via hack definition)
+            # Check for string types. (py2: unicode, py3: str via type hack above)
+            # or string subclasses. Exclude string subclasses symbol and char.
+            if isinstance(k, unicode) and not (type(k) is symbol or type(k) is char):
+                # Convert string subclasses to string
+                if not type(k) is unicode:
+                    self.properties[unicode(k)] = self.properties.pop(k)
                 continue
-            # If key is binary then change to string
-            elif isinstance(k, str):
-                # py2 str
-                self.properties[k.encode('utf-8')] = self.properties.pop(k)
+            # If key is binary then change to string. Exclude bytes subclass decimal128.
+            # Mostly for py2 users who encode strings without using the u'' prefix
+            # but py3 bytes() type will be converted also
+            elif isinstance(k, bytes) and not (type(k) is decimal128):
```

Comment:
I think you can leave this test as 'str' as it was before - it only covers the py2 str (binary) case as py3 str is already covered above in unicode, I guess you still need the decimal128 case still though.

Reply:
Good point. Agreed.